File: duolingo_rum/Models/WordProgress.cs
using System;
using System.Collections.Generic;

namespace duolingo_rum.Models;

public partial class WordProgress
{
    public int Id { get; set; }

    public Guid UserId { get; set; }

    public int WordId { get; set; }

    public int? Repetitions { get; set; }

    public decimal? EasinessFactor { get; set; }

    public int? IntervalDays { get; set; }

    public DateOnly? NextReview { get; set; }

    public DateOnly? LastReview { get; set; }

    public int? CorrectCount { get; set; }

    public int? WrongCount { get; set; }

    public bool? IsLearned { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual User User { get; set; } = null!;

    public virtual Word Word { get; set; } = null!;
}

File: android-app/app/src/main/java/com/linguaai/app/models/WordProgress.kt
package com.linguaai.app.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class WordProgress(
    val id: Int = 0,
    @SerialName("user_id")
    val userId: String = "",
    @SerialName("word_id")
    val wordId: Int = 0,
    val repetitions: Int? = 0,
    @SerialName("easiness_factor")
    val easinessFactor: Double? = null,
    @SerialName("interval_days")
    val intervalDays: Int? = null,
    @SerialName("next_review")
    val nextReview: String? = null,
    @SerialName("last_review")
    val lastReview: String? = null,
    @SerialName("correct_count")
    val correctCount: Int? = 0,
    @SerialName("wrong_count")
    val wrongCount: Int? = 0,
    @SerialName("is_learned")
    val isLearned: Boolean? = false,
    @SerialName("created_at")
    val createdAt: String? = null,
    @SerialName("updated_at")
    val updatedAt: String? = null
)
